client/src/helpers/api.ts

```diff
@@ -361,31 +361,37 @@ const updateServiceRequestStatus = async (
 
   let notes: any[] = get(serviceRequest, 'note', []).filter((n) => n.text != null);
 
-  if (note != null) {
+  if (note && note.length > 0) {
     notes = [
       ...notes,
       {
         authorReference: {
-          reference: `Practitioner/${user.practitionerId}`,
+          reference: `Practitioner/${user.practitionerData.practitioner.id}`,
         },
         text: note,
         time: new Date(),
       },
     ];
   }
 
-  const editedServiceRequest = {
+  let editedServiceRequest = {
     ...serviceRequest,
     performer: [
       {
         reference: `PractitionerRole/${user.practitionerData.practitionerRole.id}`,
       },
     ],
     extension,
-    note: notes,
     status,
   };
 
+  if (notes && notes.length > 0) {
+    editedServiceRequest = {
+      ...editedServiceRequest,
+      note: notes
+    }
+  }
+
```

Comment:
pourquoi créer un clone object?

```
const editedServiceRequest = {
     ...serviceRequest,
     performer: [
       {
         reference: `PractitionerRole/${user.practitionerData.practitionerRole.id}`,
       },
     ],
     extension,
     status,
     note: (notes && notes.length > 0) ? notes : []
   };

```

Reply:
FHIR doesn't allow note to be empty array, if empty then should not be in the request payload.

Reply:
ok, why not juste adding the properties to the const?

```
if (notes && notes.length > 0) {
     editedServiceRequest['note'] = notes
}
```

You can still use `const` since it's the same object.